### PR TITLE
Final fixes to support EL6/7

### PR DESCRIPTION
**Pull request by armab**
Final fixes https://github.com/StackStorm/ansible-st2/issues/5 to enable EL6/7 support.

- [x] Allow EL builds to show failure in Travis
- [x] EL6
  - [x] Test Under `centos/6` Vagrant
  - [x] Check Idempotence
  - [x] Fix CentOS6 build (Docker)
    - [x] BUG Fix SELinux Ansible task failure via 0c21712b33cc8762f362e893d1a6a8d2e9ef573c
    - [x] BUG [`Starting st2api: [FAILED]`](https://travis-ci.org/StackStorm/ansible-st2/jobs/194558665) via https://github.com/StackStorm/ansible-st2/pull/96/commits/c81c37d42eaa4e90375a02220a373eaa01ea1882
- [x] EL7
    - [x] Test Under `centos/7` Vagrant
    - [x] Check Idempotence
    - [x] Fix CentOS7 build (Docker)
      - [x] BUG [`cp: cannot stat ‘/usr/share/doc/st2/conf/nginx/st2.conf’: No such file or directory`](https://travis-ci.org/StackStorm/ansible-st2/jobs/194558666) via https://github.com/StackStorm/ansible-st2/pull/96/commits/c522fb4c168a0456de5e61093fdd64877f748e2f
- [x] ~BUG [`Ubuntu14/16` build](https://travis-ci.org/StackStorm/ansible-st2/jobs/194858969) -  `EPEL` dependency role is included several times, even with `when: ansible_os_family == 'RedHat'`~ (not a bug, that's how Ansible dependency works :/)
- [x] Minor improvements
   - [x] `st2web` shouldn't restart `st2`
   - [x] add `restart` and `reload` handlers for `nginx` role
   - [x] `st2web` to use `nginx restart` handler
   - [x] don't run 2 builds in Travis for every PR
- [x] Cosmetic changes
  - [x] Add missing tags
  - [x] Use consistent task splitting approach `{{ ansible_pkg_mgr }}` for all roles 
- [x] Update `meta.yml`
- [x] Update `README`

### CODE REVIEW

File: roles/nginx/tasks/nginx_yum.yml [hunk 1]
```diff
@@ -29,6 +29,14 @@
...
     name: libsemanage-python, libselinux-python
     state: present
+  register: nginx_selinux_dependencies
+  tags: nginx
+
+- name: Update SELinux facts after installing dependencies
+  become: yes
+  setup:
+    filter: ansible_selinux
+  when: nginx_selinux_dependencies.changed
```
Comment (humblearner) [inline roles/nginx/tasks/nginx_yum.yml]: 😄.. Just for reference this fixes: ```TASK [nginx : Adjust SELinux to allow network access for nginx] ****************
task path: /ansible-st2/roles/nginx/tasks/nginx_yum.yml:34
fatal: [centos6]: FAILED! => {
    "failed": true,
    "msg": "The conditional check '(ansible_selinux.status == \"True\")' failed. The error was: error while evaluating conditional ((ansible_selinux.status == \"True\")): 'bool object' has no attribute 'status'\n\nThe error appears to have been in '/ansible-st2/roles/nginx/tasks/nginx_yum.yml': line 34, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Adjust SELinux to allow network access for nginx\n  ^ here\n"
}```

**Comment by armab** [comment]
Holy cow, `EL7` build is passing: https://github.com/StackStorm/ansible-st2/pull/96
Now only `EL6` left with error I can't reproduce at all, happens only in Travis env :/

```
failed: [localhost] (item=st2api) => {"failed": true, "item": "st2api", "msg": "Starting st2api: [FAILED]\r\n"}
```
Needs more debugging.

**Comment by armab** [comment]
More light about `st2api` `st2auth` `st2stream` failed services under `EL6`:
https://github.com/StackStorm/ansible-st2/issues/99

**Comment by armab** [comment]
Regarding
> BUG [`Ubuntu14/16` build](https://travis-ci.org/StackStorm/ansible-st2/jobs/194858969) -  `EPEL` dependency role is included several times, even with `when: ansible_os_family == 'RedHat'

So it's not a bug, that's how Ansible dependency works, sadly.
[This construction](https://github.com/StackStorm/ansible-st2/blob/8c8bcdbca61c24f942b32ae2489ac7e7537609b9/roles/mongodb/meta/main.yml#L20-L21) just appends `when:` to every task in dependant role, but does not prevent the role from beeing loaded.
See: https://github.com/ansible/ansible/issues/15254

cc @Mierdin ^^ you might be interested.

File: roles/nginx/tasks/nginx_yum.yml (hunk 1 above)
Comment (Mierdin) [inline roles/nginx/tasks/nginx_yum.yml]: 👍 I was just about to report this issue; came up in my CentOS testing.

File: roles/nginx/handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+- name: restart nginx
+  become: yes
+  service:
+    name: nginx
+    state: restarted
+
+- name: reload nginx
```
Comment (Mierdin) [inline roles/nginx/handlers/main.yml]: I only found references for the `restart` handler. Is this `reload` handler meant to be used anywhere?

File: roles/st2/tasks/user.yml [hunk 3]
```diff
@@ -10,7 +10,6 @@
     ssh_key_file: "{{ st2_ssh_key_file }}"
     state: present
   register: _user
-  tags: [st2, user]
```
Comment (Mierdin) [inline roles/st2/tasks/user.yml]: No tags at all? I can see not using `st2`, but the other tasks in this role have a tag; was just curious why this is being removed entirely

Comment (armab) [inline roles/st2/tasks/user.yml]: I moved `tags` for this role into `tasks/main.yml`, see: https://github.com/StackStorm/ansible-st2/blob/4b9cb5188a34f8cde587a10c57401078c947af51/roles/st2/tasks/main.yml

So tags are applied there, - a bit easier and tags still can be used.

File: roles/nginx/handlers/main.yml (hunk 2 above)
Comment (armab) [inline roles/nginx/handlers/main.yml]: Not used right now but brings just a bit more complete role.
Still can be used in future and in our external infra where we just include `nginx` role and use it how we like it.
Ex: install nginx in internal infra, same for `mongo`, `postgres` or any other role from this repo.

**Pull request merged**